components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -231,6 +245,15 @@ private String fetchRelativeUrl() throws URLBuilderException {
             appendParamsToUri(relativeUrl, fragment, "#");
             return relativeUrl.toString();
         }
+
+        private String fetchRelativeInternalUrl() throws URLBuilderException {
+
+            StringBuilder relativeUrl = new StringBuilder();
+            String resolvedParamsString = getResolvedParamString(parameters);
```

Comment:
Don't we need to append the url context here?

Reply:
no